Developing-an-Interactive-Educational-Platform-with-Directus.md

```diff
@@ -0,0 +1,374 @@
+
+-   View All Data: Instructors should have access to view all data within the platform. Restricted 
+
+-   Limited administrative access: Instructors should not have access to manage users, roles, or other administrative settings.
+
+![](https://lh7-us.googleusercontent.com/YGWsatZncTynyPnj8p5a9ANJzu3irWvKQ4S3btPUr6GUPJTIfHBLKYXhfSCwJuV8_kkHwNopI2mkRALEsYuik--jck8gLalu3-CGJrPFz_LGh4fqV290UCJs0TYEHFtY5ZKp0L8AB-tKJ3M2KdJOFio)
+
+### Custom Permissions
+
+![](https://lh7-us.googleusercontent.com/RHP_HpUinE5HzuVJOPQBYXrx1kyZJRw04Qil8wL3CmnJiuBTM_AO-lWkFtcRfpALgmhYPBoRKikjrAF6vModJfcROIuWLJjE07TXJEw3ty01X76KO-sG5VGtNRNAvg7eUIyDSDrC4IKmbDtzVRSihzE)
+
+Now for the instructors, since they do have a little more hedge of control, we would want to insist on them having limited rights to do certain things, and there's where custom permissions come into play, Directus in its mass quality didn't leave tables empty on this one, with Directus you could easily restrict certain rights as we earlier mentioned.
+
+We do want instructors to create a course quite alright, but we also would want them not to update other instructors' courses.
+
+So in the instructors roles, we will go instructor-courses collection, click edit, select  custom item, go to rules, click the filter drop down, click instructor -Id then set it to $CURRENT_USER.id. This should make only instructors who created certain courses be able to edit them.
+
+![](https://lh7-us.googleusercontent.com/eszZASHeqR49PsGeZQF7dJwKeJ-2R27676PFkVcFGlZKLbe-xRoCrgBZW3lXTDBfLjjpATpWINEYTQRdFGToRD9zmQTefOwpDufFvWVyYNgQGkQThs6F6FZPR1Sj7uba5CFA6CHWSVUJOyDAm3ddEUk)
+
+You thought we had finished, but not quite right, we do need custom permissions for the students, as we only want students to view published courses, lessons, and quizzes.
+
+We will open up the read permission click custom permissions, and under item permissions, we wiil create a role using a dynamic variable, by clicking the add filter drop dropdown, we are going to go to the quiz questions field, and use the ' Quiz questions to equals $CURRENT_USER.id", and now they would only be able to see the quizzes that belongs to them alone.
+
+![](https://lh7-us.googleusercontent.com/TsilbdWhcx2I1Oy_uNME4cO-BfJMBcHWmyNCZBtj-obb5NGCnfBsrFrSsP8stVrfz12xI2jm1MXIlK4W0ZhX4nk31YJL62EFbie0f59NYZe4K659T3gi5skIkA3FoT5ZN20jtjOJM87MWAgr8-6Q-OI)
+
+Lastly, we only want students to see published courses, or lessons; we navigate to the courses collection, the custom permissions, then item permissions, then we click the add filter dropdown under the rules, select status and let it be equal to published. And viola students can only see published courses, we will go ahead to do same for, lessons, and quiz.
+
+![](https://lh7-us.googleusercontent.com/hZBD0NEPySG1pe8Y1hQzGAcf8h-QQJvr9qoEDiphvUaX6p0Mkag2TRs-KRPTOIVMYTlLMBjLZoUejxlv4a5SlIBPwtqLc19qXH943GJs3-bkd8ZbdggqOBi7J8K16seeEv0BLlQuoqJ3WUnqvlOPvbg)
```

Comment:
Please put the custom permissions inline, not at the end. For example, you say "Students should be able to view assignments within the courses they are enrolled in." and then show a screenshot where they have "all" access.